.changeset/eager-rooms-turn.md

```
@@ -0,0 +1,5 @@
+---
+'@mastra/mcp': patch
+---
+
+Preserve structured HTTP status and transport codes in aggregate MCP discovery errors while retaining legacy string errors.
```

markdownlint-cli2 (0.23.2)

[warning] First line in a file should be a top-level heading

(MD041, first-line-heading, first-line-h1)